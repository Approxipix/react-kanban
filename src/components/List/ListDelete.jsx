import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteList } from '../../redux/listReducer/actions';
import { Text, SubmitButton } from '../BaseComponent';

class ListDelete extends Component {
  deleteList = () => {
    const { listId, boardId } = this.props;
    this.props.actions.deleteList({
      boardId: boardId,
      listId: listId,
    });
  };

  render() {
    return (
      <>
        <Text>
          All actions will be removed from the activity feed and you wonâ€™t be able to re-open the list.
          There is no undo.
        </Text>
        <SubmitButton
          color={'danger'}
          onClick={() => this.deleteList()}
        >
          Delete list
        </SubmitButton>
      </>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteList: deleteList,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(ListDelete);
