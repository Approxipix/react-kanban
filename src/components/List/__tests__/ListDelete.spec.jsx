import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ListDelete from '../ListDelete';

const mockStore = configureStore([]);

describe('<ListDelete>', () => {
  let store;
  let component;
  let mockDeleteList = jest.fn();

  beforeEach(() => {
    store = mockStore();
    store.dispatch = mockDeleteList;
    const props = {
      boardId: 'BoardID',
      listId: 'ListID',
    };

    const wrapper = shallow(<ListDelete store={store} {...props} />);
    component = wrapper.find('ListDelete').dive();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should dispatch an action after click on button", () => {
    const expectedAction = {
      type: 'DELETE_LIST',
      payload: {
        boardId: 'BoardID',
        listId: 'ListID',
      }
    };

    component.find('SubmitButton').simulate('click');
    expect(mockDeleteList).toHaveBeenCalledTimes(1);
    expect(mockDeleteList).toHaveBeenCalledWith(expectedAction);
  });
});
