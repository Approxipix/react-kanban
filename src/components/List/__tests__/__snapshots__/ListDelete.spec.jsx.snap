// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListDelete> should render with given state from Redux store 1`] = `
<Fragment>
  <styled.p>
    All actions will be removed from the activity feed and you wonâ€™t be able to re-open the list. There is no undo.
  </styled.p>
  <SubmitButton
    color="danger"
    onClick={[Function]}
  >
    Delete list
  </SubmitButton>
</Fragment>
`;
